Code/C++/Doubly_linkedlist/Doubly_linkedlist_all_operations.cpp

```diff
@@ -0,0 +1,390 @@
+In this program you will be given a Menu and you have to choose which operation You want to perform on the Doubly link list. 
+According to your choice the program will perform the operation on the Doubly link list.*/
+
+#include <iostream>
+#include <cstdlib>
+using namespace std;
+
+//Basic structure of a node in doubly linked list consisting of value and two pointers of the node.
+class node
+{
+public:
+    int data;
+    node *next;
+    node *prev;
+};
+class Doubly_linklist
+{
+    node *start;
+    node *last;
+
+public:
+    Doubly_linklist()
+    {
+        start = nullptr;
+        last = nullptr;
+    }
+    void creation();
+    void traversal();
+    void insertion_at_beginning();
+    void insertion_at_end();
+    void insertion_after_given_index();
+    void insertion_at_index();
+    void deletion_at_beginning();
+    void deletion_at_end();
+    void deletion_at_index();
+};
+
+//Function for creating a new node.
+void Doubly_linklist::creation()
+{
+    node *new_node = new node;
+    cout << "\nEnter the info of the node ";
+    cin >> new_node->data;
+    new_node->next = nullptr;
+    new_node->prev = nullptr;
+    if (start == nullptr)
+    {
+        start = last = new_node;
+    }
+    else
+    {
+        last->next = new_node;
+        new_node->prev = last;
+        last = new_node;
+    }
+}
+
+//Function for visiting each node once in the list.
+void Doubly_linklist::traversal()
+{
+    node *ptr = start;
+    node *p = last;
+    int count = 0;
+    if (ptr == nullptr|| p == nullptr)
+    {
+        cout << "List is empty " << endl;
+    }
+    else
+    {
+        cout << "\ntraversing from left from right ";
+        while (ptr != nullptr)
+        {
+            cout << "\nInfo of node at index " << count << "--> " << ptr->data;
+            ptr = ptr->next;
+            count += 1;
+        }
+        count -= 1;
+        cout << "\nTransversing from right to left ";
+        while (p != nullptr)
+        {
+            cout << "\nInfo of node at index " << count << "--> " << p->data;
+            p = p->prev;
+            count -= 1;
+        }
+    }
+}
+
+//Function for inserting a node in beginning.
+
+void Doubly_linklist::insertion_at_beginning()
+{
+    node *new_node = new node;
+    cout << "\nEnter the info of the node ";
+    cin >> new_node->data;
+    if (start == nullptr && last == nullptr)
+    {
+        new_node->prev = nullptr;
+        new_node->next = nullptr;
+        start = last = new_node;
+    }
+    else
+    {
+        new_node->prev = nullptr;
+        new_node->next = start;
+        start->prev = new_node;
+        start = new_node;
+    }
+}
+
+//Function for inserting new node at end of list.
+
+void Doubly_linklist::insertion_at_end()
+{
+    node *new_node = new node;
+    cout << "\nEnter the info of node ";
+    cin >> new_node->data;
+    if (last == nullptr)
+    {
+        cout << "\nList is empty ";
+    }
```

Comment:
This case will be the same as creating the first node because adding a node to the last of an empty list is the same as creating its first node.